### ISSUE TITLE
emit error event when the socket is denied by the server.

### ISSUE THREAD

**Issue by chirag04** [socketio/socket.io-client-swift#45]
We have server side authentication on socket.io. Incase the server rejects the connection, socket.io client should receive ``error`` event or some other event to handle that.

Currently it's not handling this case and the app crashes everytime the socket is denied.

**Comment by nuclearace** [comment]
Can you provide a stack trace of the crash? Also, how is the socket being rejected? Is the authentication being done in the first handshake?

**Comment by chirag04** [comment]
Yes authentication is done in first handshake. Some like this:

```
io.use(function(socket, next) {
     if(false) return next(err);
});
```

I can send you the stack trace in some time.

**Comment by nuclearace** [comment]
https://github.com/socketio/socket.io-client-swift/compare/engineerrors to track changes

**Comment by nuclearace** [comment]
Got a fix coming along

**Comment by nuclearace** [comment]
@chirag04 this should be fixed in v2.0.1

**Comment by chirag04** [comment]
sweet. I will test it and let you know.
